app/models/mvi.rb

```diff
@@ -121,18 +121,23 @@ def mvi_response
 
   def response_from_redis_or_service
     do_cached_with(key: @user.uuid) do
-      profile = mvi_service.find_profile(@user)
-      self.class.redis_namespace_ttl = set_ttl(profile)
-      profile
+      mvi_service.find_profile(@user)
     end
   end
 
   def mvi_service
     @service ||= MVI::Service.new
   end
 
-  def set_ttl(profile)
-    if profile.status == MVI::Responses::FindProfileResponse::RESPONSE_STATUS[:ok]
+  def save
```

Comment:
might be helpful to have a comment that this is overriding the base class behavior -- agree with Bill that the duplicate code is a bit of a smell, but not a show stopper at all.